src/gtest/test_joinsplit.cpp

```diff
@@ -587,3 +587,82 @@ TEST(Joinsplit, NoteClass)
     ASSERT_EQ(note.r, clone.r);
     ASSERT_EQ(note.a_pk, clone.a_pk);
 }
+
+TEST(Joinsplit, BasicJoinsplitVerification)
+{
+    LoadProofParameters();
+
+    // We only check that joinsplits are constructed properly
+    // and verify them here.
+    //
+    // See #471, #520, #459 and probably others.
+    //
+    // There may be ways to use boost tests to catch failing
+    // threads or processes (?) but they appear to not work
+    // on all platforms and would gently push us down an ugly
+    // path. We should just fix the assertions.
+    //
+    // Also, it's generally libzcash's job to ensure the
+    // integrity of the scheme through its own tests.
```

Comment:
```suggestion
    // and verify them here. It's generally libzcash's job to
    // ensure the integrity of the scheme through its own tests.
```
#471, #520, and #459 are all long fixed; they were inappropriate error handling in libsnark and libzerocash (sic). The counterpart of this comment in [src/test/transaction_tests.cpp](https://github.com/zcash/zcash/pull/5664/files#diff-cc0c6a9039a1c9fe38b8a21fe28391fffbac9b8531dfda0f658919a9f74b46ba) has actually been stale since before launch, and #4246 should have removed it as well.